File: TOMIMSV4/Entity/Transacciones/OrdenCompra/OC_Detalle/clsBeTrans_oc_det.vb
Public Class clsBeTrans_oc_det
    Implements ICloneable
    Implements IDisposable

    Public Property IdOrdenCompraEnc() As Integer = 0
    Public Property IdOrdenCompraDet() As Integer = 0
    Public Property IdProductoBodega() As Integer = 0
    Public Property IdArancel() As Integer = 0
    Public Property IdPresentacion() As Integer = 0
    Public Property IdUnidadMedidaBasica() As Integer = 0
    Public Property IdMotivoDevolucion() As Integer = 0
    Public Property No_Linea() As Integer = 0
    Public Property Nombre_producto() As String = ""
    Public Property Nombre_presentacion() As String = ""
    Public Property Nombre_arancel() As String = ""
    Public Property Porcentaje_arancel() As Double = 0.0
    Public Property Nombre_unidad_medida_basica() As String = ""
    Public Property Cantidad() As Double = 0.0
    Public Property Peso As Double = 0.0
    Public Property Peso_Recibido As Double = 0.0
    Public Property Cantidad_recibida() As Double = 0.0
    ''' <summary>
    ''' Costo unitario
    ''' </summary>
    ''' <returns></returns>
    Public Property Costo() As Double = 0.0
    Public Property Total_linea() As Double = 0.0
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False
    Public Property Atributo_variante_1() As String = ""
    Public Property Codigo_Producto As String = ""

    '#EJC20201228: Agregué campos necesarios poliza.
    Public Property valor_aduana As Double = 0
    Public Property valor_fob As Double = 0
    Public Property valor_iva As Double = 0
    Public Property valor_dai As Double = 0
    Public Property valor_seguro As Double = 0
    Public Property valor_flete As Double = 0
    Public Property Peso_Neto As Double = 0
    Public Property Peso_Bruto As Double = 0

    '#EJC20210313: Agregado para consolidadores de CEALSA
    Public Property IdPropietarioBodega As Integer = 0

    '#EJC20210322: Agregué porque creo que se utilizará para la gestión de documentos en tránsito...
    Public Property IdPedidoCompraDet As Integer = 0

    '#EJC20210403:Guarda el IdDet del detalle de la OC que pertenece al producto que es de tipo Kit o padre.
    Public Property IdOrdenCompraDetPadre As Integer = 0

    '#EJC20220224: Cealsa, guardar el shipper o embarcador 
    Public Property IdEmbarcador As Integer = 0
    Public Property Nombre_Embarcador As String = ""
    Public Property Nombre_Clasificacion As String = ""
    Public Property IdProductoTallaColor As Integer = 0

    Public Property Camas_Tarima As Integer = 0

    Public Property Cajas_Cama As Integer = 0

    Sub New()
    End Sub

    Sub New(ByRef IdOrdenCompraEnc As Integer, ByVal IdOrdenCompraDet As Integer, ByVal IdProductoBodega As Integer, ByVal IdArancel As Integer, ByVal IdPresentacion As Integer, ByVal IdUnidadMedidaBasica As Integer, ByVal IdMotivoDevolucion As Integer, ByVal No_Linea As Integer, ByVal nombre_producto As String, ByVal nombre_presentacion As String, ByVal nombre_arancel As String, ByVal porcentaje_arancel As Double, ByVal nombre_unidad_medida_basica As String, ByVal cantidad As Double, ByVal cantidad_recibida As Double, ByVal costo As Double, ByVal total_linea As Double, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal activo As Boolean)
        Me.IdOrdenCompraEnc = IdOrdenCompraEnc
        Me.IdOrdenCompraDet = IdOrdenCompraDet
        Me.IdProductoBodega = IdProductoBodega
        Me.IdArancel = IdArancel
        Me.IdPresentacion = IdPresentacion
        Me.IdUnidadMedidaBasica = IdUnidadMedidaBasica
        Me.IdMotivoDevolucion = IdMotivoDevolucion
        Me.No_Linea = No_Linea
        Me.Nombre_producto = nombre_producto
        Me.Nombre_presentacion = nombre_presentacion
        Me.Nombre_arancel = nombre_arancel
        Me.Porcentaje_arancel = porcentaje_arancel
        Me.Nombre_unidad_medida_basica = nombre_unidad_medida_basica
        Me.Cantidad = cantidad
        Me.Cantidad_recibida = cantidad_recibida
        Me.Costo = costo
        Me.Total_linea = total_linea
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.Activo = activo
    End Sub

    Public Function Clone() As Object Implements ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Trans_oc/clsBeTrans_oc_det.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;
using WMS.EntityCore.Producto;
using WMSWebAPI.Be;

namespace WMS.EntityCore.Trans_oc
{
    public class clsBeTrans_oc_det : ICloneable
    {
        [Column("IdOrdenCompraEnc")]
        [DisplayName("IdOrdenCompraEnc")]
        public int IdOrdenCompraEnc { get; set; } = 0;

        [Column("IdOrdenCompraDet")]
        [DisplayName("IdOrdenCompraDet")]
        public int IdOrdenCompraDet { get; set; } = 0;

        [Column("IdProductoBodega")]
        [DisplayName("IdProductoBodega")]
        public int IdProductoBodega { get; set; } = 0;

        [Column("IdArancel")]
        [DisplayName("IdArancel")]
        public int IdArancel { get; set; } = 0;

        [Column("IdPresentacion")]
        [DisplayName("IdPresentacion")]
        public int IdPresentacion { get; set; } = 0;

        [Column("IdUnidadMedidaBasica")]
        [DisplayName("IdUnidadMedidaBasica")]
        public int IdUnidadMedidaBasica { get; set; } = 0;

        [Column("IdMotivoDevolucion")]
        [DisplayName("IdMotivoDevolucion")]
        public int IdMotivoDevolucion { get; set; } = 0;

        [Column("No_Linea")]
        [DisplayName("No_Linea")]
        public int No_Linea { get; set; } = 0;

        [Column("nombre_producto")]
        [DisplayName("nombre_producto")]
        public string Nombre_producto { get; set; } = "";

        [Column("nombre_presentacion")]
        [DisplayName("nombre_presentacion")]
        public string Nombre_presentacion { get; set; } = "";

        [Column("nombre_arancel")]
        [DisplayName("nombre_arancel")]
        public string Nombre_arancel { get; set; } = "";

        [Column("porcentaje_arancel")]
        [DisplayName("porcentaje_arancel")]
        public double Porcentaje_arancel { get; set; } = 0;

        [Column("nombre_unidad_medida_basica")]
        [DisplayName("nombre_unidad_medida_basica")]
        public string Nombre_unidad_medida_basica { get; set; } = "";

        [Column("cantidad")]
        [DisplayName("cantidad")]
        public double Cantidad { get; set; } = 0;

        [Column("cantidad_recibida")]
        [DisplayName("cantidad_recibida")]
        public double Cantidad_recibida { get; set; } = 0;

        [Column("costo")]
        [DisplayName("costo")]
        public double Costo { get; set; } = 0;

        [Column("total_linea")]
        [DisplayName("total_linea")]
        public double Total_linea { get; set; } = 0;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("peso")]
        [DisplayName("peso")]
        public double Peso { get; set; } = 0;

        [Column("peso_recibido")]
        [DisplayName("peso_recibido")]
        public double Peso_recibido { get; set; } = 0;

        [Column("atributo_variante_1")]
        [DisplayName("atributo_variante_1")]
        public string Atributo_variante_1 { get; set; } = "";

        [Column("codigo_producto")]
        [DisplayName("codigo_producto")]
        public string Codigo_producto { get; set; } = "";

        [Column("valor_aduana")]
        [DisplayName("valor_aduana")]
        public double Valor_aduana { get; set; } = 0;

        [Column("valor_fob")]
        [DisplayName("valor_fob")]
        public double Valor_fob { get; set; } = 0;

        [Column("valor_iva")]
        [DisplayName("valor_iva")]
        public double Valor_iva { get; set; } = 0;

        [Column("valor_dai")]
        [DisplayName("valor_dai")]
        public double Valor_dai { get; set; } = 0;

        [Column("valor_seguro")]
        [DisplayName("valor_seguro")]
        public double Valor_seguro { get; set; } = 0;

        [Column("valor_flete")]
        [DisplayName("valor_flete")]
        public double Valor_flete { get; set; } = 0;

        [Column("peso_neto")]
        [DisplayName("peso_neto")]
        public double Peso_neto { get; set; } = 0;

        [Column("peso_bruto")]
        [DisplayName("peso_bruto")]
        public double Peso_bruto { get; set; } = 0;

        [Column("IdPropietarioBodega")]
        [DisplayName("IdPropietarioBodega")]
        public int IdPropietarioBodega { get; set; } = 0;

        [Column("nombre_propietario")]
        [DisplayName("nombre_propietario")]
        public string Nombre_propietario { get; set; } = "";

        [Column("IdOrdenCompraDetPadre")]
        [DisplayName("IdOrdenCompraDetPadre")]
        public int IdOrdenCompraDetPadre { get; set; } = 0;

        [Column("IdEmbarcador")]
        [DisplayName("IdEmbarcador")]
        public int IdEmbarcador { get; set; } = 0;
        public clsBeProducto_presentacion Presentacion { get; set; } = new clsBeProducto_presentacion();
        public bool IsNew { get; set; }
        public bool ExisteEnRecepcion { get; set; }
        public decimal FactorPresentacion { get; set; }
        public clsBeArancel Arancel { get; set; } = new clsBeArancel();
        public clsBeProducto Producto { get; set; } = new clsBeProducto();
        public clsBeUnidad_medida UnidadMedida { get; set; } = new clsBeUnidad_medida();
        public int RowIndex { get; set; } = 0;
        public List<clsBeTrans_oc_det> lProductosHijosKit { get; set; } = new List<clsBeTrans_oc_det>();
        public string Nombre_Propietario { get; set; } = "";
        public clsBeTalla Talla { get; set; } = new clsBeTalla();
        public clsBeColor Color { get; set; } = new clsBeColor();
        public int IdProductoTallaColor { get; set; } = 0;
        public string Nombre_Embarcador { get; set; } = "";
        public string Nombre_Clasificacion { get; set; } = "";
        public int Camas_Tarima { get; set; } = 0;
        public int Cajas_Cama { get; set; } = 0;
        public clsBeTrans_oc_det() { }
        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}